src/lib/server/route-matcher.ts

```diff
@@ -0,0 +1,39 @@
+import type { Autocomplete } from '@clerk/types'
+import { pathToRegexp } from 'path-to-regexp'
+
+type WithPathPatternWildcard<T = string> = `${T & string}(.*)`
+
+type RouteMatcherRoutes = Autocomplete<WithPathPatternWildcard>
+
+export type RouteMatcherParam = Array<RegExp | RouteMatcherRoutes> | RegExp | RouteMatcherRoutes
+
+/**
+ * Returns a function that accepts a `Request` object and returns whether the request matches the list of
+ * predefined routes that can be passed in as the first argument.
+ *
+ * You can use glob patterns to match multiple routes or a function to match against the request object.
+ * Path patterns and regular expressions are supported, for example: `['/foo', '/bar(.*)'] or `[/^\/foo\/.*$/]`
+ * For more information, see: https://clerk.com/docs
+ */
+export const createRouteMatcher = (routes: RouteMatcherParam) => {
```

Comment:
This is the same helper function exported by the [Nextjs SDK](https://clerk.com/docs/references/nextjs/clerk-middleware#create-route-matcher) that helps matching routes to be protected